BasicMusicPlayer/app/src/main/java/playback/AudioPlaybackService.kt

```diff
@@ -251,23 +248,64 @@ class AudioPlaybackService : Service() {
         LocalBroadcastManager.getInstance(this).sendBroadcast(intent)
     }
 
-    fun muteAudio() {
-        println("made it to mute")
+    @RequiresApi(Build.VERSION_CODES.O)
+    private fun muteAudio() {
         mediaPlayer?.setVolume(0f, 0f)
         isMuted = true
+        if (::audioFocusRequest.isInitialized){
+            audioManager.abandonAudioFocusRequest(audioFocusRequest)
+        }
     }
 
-    fun unmuteAudio() {
-        println("made it to unmute")
-        mediaPlayer?.setVolume(1f, 1f)
-        isMuted = false
+    @RequiresApi(Build.VERSION_CODES.O)
+    private fun unmuteAudio() {
+        if (::audioFocusRequest.isInitialized){
+            println("audio focus request is said to be initialized")
+            val focusRequestResult = audioManager.requestAudioFocus(audioFocusRequest)
+            if (focusRequestResult == AudioManager.AUDIOFOCUS_REQUEST_GRANTED) {
+                println("WE GOT THE AUDIO")
+                mediaPlayer?.setVolume(1f, 1f)
+                isMuted = false
+            }
+            else{
+                println("wAH WWAH")
```

Comment:
So this is fine, but could you add something to indicate the condition that's being logged here so that others understand it when they see this in the logs?